src/itwinai/torch/loggers.py

```diff
@@ -0,0 +1,238 @@
+            return
+
+        # Log checkpoints if the last checkpoint was saved but not logged
+        if self._checkpoint_callback:
+            self._scan_and_log_checkpoints(self._checkpoint_callback)
+
+        self.experiment.destroy_logger_context()
+
+    @override
+    @rank_zero_only
+    def log_metrics(
+            self,
+            metrics: Dict[str, float],
+            step: Optional[int] = None
+    ) -> None:
+        """Lightning Logger function.
+        Logs the given metrics and is usually called by the Lightning Trainer.
+
+        Args:
+            metrics (Dict[str, float]): Dictionary of metrics to log.
+            step (Optional[int], optional): Training step associated with the metrics.
+            Defaults to None.
+        """
+        for key, value in metrics.items():
+            self.experiment.log(
+                item=value,
+                identifier=key,
+                kind='metric',
+                step=step
+            )
+
+    @override
+    @rank_zero_only
+    def log_hyperparams(
+            self,
+            params: Union[Dict[str, Any], Namespace]
+    ) -> None:
+        """Lightning Logger function. Logs hyperparameters for the experiment.
+
+        Args:
+            params (Union[Dict[str, Any], Namespace]): Hyperparameters dictionary or object.
+        """
+        if isinstance(params, Namespace):
+            params = vars(params)
+
+        self.experiment.save_hyperparameters(params)
+
+    @override
+    def after_save_checkpoint(
+        self,
+        checkpoint_callback: ModelCheckpoint
+    ) -> None:
+        """Lightning Logger function. Handles checkpoint saving to the logger after
+        the ModelCheckpoint Callback of the Lightning Trainer is called.
+        The checkpoints are logged as artifacts.
+
+        Args:
+            checkpoint_callback (ModelCheckpoint): Callback instance to manage checkpointing.
+        """
+        if self._log_model == "all" or (
+            self._log_model is True and checkpoint_callback.save_top_k == -1
+        ):
+            self._scan_and_log_checkpoints(checkpoint_callback)
+        elif self._log_model is True:
+            self._checkpoint_callback = checkpoint_callback
+
+    def _scan_and_log_checkpoints(
+        self,
+        checkpoint_callback: ModelCheckpoint
+    ) -> None:
+        """Scans and logs checkpoints as artifacts in the experiment.
+
+        This function retrieves new checkpoints, logs them as artifacts, and saves
+        related metadata, including checkpoint score, filename, and any model checkpoint
+        parameters.
+
+        Args:
+            checkpoint_callback (ModelCheckpoint): Callback instance used for checkpointing.
+        """
+        # Get checkpoints to be saved with associated score
+        checkpoints = _scan_checkpoints(checkpoint_callback, self._logged_model_time)
+
+        # Log iteratively all new checkpoints
+        for t, p, s, tag in checkpoints:
```

Comment:
I feel like I recognize this code from something I saw in the Lightning docs, so that might be the reason, but I think you are much better off here naming the variables something else than "t", "p" etc., as I'm not sure what they mean atm

Reply:
Good point, it actually took me so long to understand this myself